### PR TITLE
tidyp: revert to fetchurl

### PR DESCRIPTION
**Pull request by grahamc**
It doesn't build with fetchFromGitHub.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/tidyp/default.nix [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
   pname = "tidyp";
   version = "1.04";
 
-  src = fetchFromGitHub {
-    owner = "petdance";
-    repo = "tidyp";
-    rev = version;
-    sha256 = "0jslskziwzk4hb6i640fvpnbv2zxrvim6pdx2gwx5wyc64aviskc";
+  src = fetchurl {
+    url = "https://github.com/downloads/petdance/tidyp/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tidyp/default.nix:8]: that URL is a 404 and only exists in cache.

Comment (grahamc) [reply pkgs/development/libraries/tidyp/default.nix:8]: Hmm... it seems to work for me? 

```
grahamc@hyperchicken:~/projects/github.com/NixOS/nixpkgs/ > nix-build /nix/store/jg3qgrbmnm1vanljm2w2q30q48h84kg6-tidyp-1.04.tar.gz.drv --check
checking outputs of '/nix/store/jg3qgrbmnm1vanljm2w2q30q48h84kg6-tidyp-1.04.tar.gz.drv'...

trying http://tarballs.nixos.org/sha256/0f5ky0ih4vap9c6j312jn73vn8m2bj69pl2yd3a5nmv35k9zmc10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  647k  100  647k    0     0   929k      0 --:--:-- --:--:-- --:--:--  929k
/nix/store/gbwl81l8rkypk47bi7y3bh70sk66rp0x-tidyp-1.04.tar.gz
```

Comment (grahamc) [reply pkgs/development/libraries/tidyp/default.nix:8]: Oh I see.

Comment (grahamc) [reply pkgs/development/libraries/tidyp/default.nix:8]: No, still seems fine:

```
grahamc@hyperchicken:~/projects/github.com/NixOS/nixpkgs/ > curl -L https://github.com/downloads/petdance/tidyp/tidyp-1.04.tar.gz --output foo
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   426  100   426    0     0    460      0 --:--:-- --:--:-- --:--:--   460
100  647k  100  647k    0     0   464k      0  0:00:01  0:00:01 --:--:-- 2900k
grahamc@hyperchicken:~/projects/github.com/NixOS/nixpkgs/ > sha256sum foo
20b0fad32c63575bd4685ed09b8c5ca222bbc7b15284210d4b576d0223f0b338  foo


grahamc@hyperchicken:~/projects/github.com/NixOS/nixpkgs/ > sha256sum /nix/store/gbwl81l8rkypk47bi7y3bh70sk66rp0x-tidyp-1.04.tar.gz
20b0fad32c63575bd4685ed09b8c5ca222bbc7b15284210d4b576d0223f0b338  /nix/store/gbwl81l8rkypk47bi7y3bh70sk66rp0x-tidyp-1.04.tar.gz

```

Comment (SuperSandro2000) [inline pkgs/development/libraries/tidyp/default.nix:8]:
```suggestion
    # downloads from a legacy GitHub download page from ~11 years ago
    # project does not work with autoconf anymore and the configure script cannot be generated from the source download
    url = "https://github.com/downloads/petdance/tidyp/${pname}-${version}.tar.gz";
```

Comment (SuperSandro2000) [reply pkgs/development/libraries/tidyp/default.nix:8]: That is a very legacy download page. The listing is under https://github.com/petdance/tidyp/downloads. Lets add a comment that we do not break this in the future.

**Comment by Stunkymonkey** [comment]
sorry for causing this much trouble.

**Comment by grahamc** [comment]
No worries, thanks for the initial PR which was largely great :)

On July 22, 2021, GitHub ***@***.***> wrote:
> sorry for causing this much trouble.
>

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/NixOS/nixpkgs/pull/130999#issuecomment-884827619>,
or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAASXLGJPXRJWWEGQK76X3TTY73KNANCNFSM5AY7JI5Q>.

**Comment by Stunkymonkey** [comment]
i will continue to make changes for `pname` and `version`, but `fetchFrom...` will be in seperate PRs.

**Pull request merged**